lib/image-stream/utils.js

```diff
@@ -16,47 +16,30 @@
 
 'use strict';
 
-const _ = require('lodash');
 const Bluebird = require('bluebird');
-const fs = Bluebird.promisifyAll(require('fs'));
-const archiveType = require('archive-type');
+const mime = require('mime-types');
 
 /**
  * @summary Get archive mime type
  * @function
  * @public
  *
- * @param {String} file - file path
+ * @param {String} filename - file path
  * @fulfil {String} - mime type
  * @returns {Promise}
  *
  * @example
- * utils.getArchiveMimeType('path/to/raspberrypi.img.gz').then((mimeType) => {
- *   console.log(mimeType);
- * });
+ * const mimeType = utils.getArchiveMimeType('path/to/raspberrypi.img.gz');
  */
-exports.getArchiveMimeType = (file) => {
-
-  // `archive-type` only needs the first 261 bytes
-  // See https://github.com/kevva/archive-type
-  const ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH = 261;
-
-  return Bluebird.using(fs.openAsync(file, 'r').disposer((fileDescriptor) => {
-    return fs.closeAsync(fileDescriptor);
-  }), (fileDescriptor) => {
-    const BUFFER_START = 0;
-    const chunk = new Buffer(ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH);
+exports.getArchiveMimeType = (filename) => {
+  // NOTE: For some awkward reason `mime-db` (used by `mime-types`)
+  // does not contain a file extension for `.gz`, so we need to test for it here
+  // TODO: Remove this once https://github.com/jshttp/mime-db/pull/68 is merged
+  if (/\.gz$/i.test(filename)) {
+    return 'application/gzip';
+  }
 
-    return fs.readAsync(
-      fileDescriptor,
-      chunk,
-      BUFFER_START,
-      ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH,
-      null
-    ).then(() => {
-      return _.get(archiveType(chunk), [ 'mime' ], 'application/octet-stream');
-    });
-  });
+  return mime.lookup(filename) || 'application/octet-stream';
```

Comment:
I initially agreed with @jhermsmeier that we ought to check `mime-types` first and `file-type` second, but I decided to have a look at this in a bit more detail...

As @jhermsmeier points out, `archive-type` is just a slim wrapper around `file-type` - with the difference being that `archive-type` only needs 261 bytes (although see https://github.com/kevva/archive-type/issues/7), and can only identify "archive" file types (7z, bz2, gz, rar, tar, zip, xz); whereas `file-type` needs the first 4100 bytes, but can identify a much larger range of file types.
Up until now, using just `archive-type` has sufficed for Etcher, since it's only "archive" types that we need to do any special processing on. However as @jhermsmeier points out, even `file-type` is unlikely to be able to identify DMGs, and we _do_ need to do special processing on DMGs, which is why he's switched to using `mime-types`.

So IMHO we should stick with using `archive-type` rather than `file-type`, and given that restriction (i.e. that archive-type doesn't even attempt to identify DMGs) I think it probably makes sense to go with @jviotti 's suggestion of checking with `archive-type` first, and only if that returns `false` proceeding with checking via `mime-types`. And I believe if `mime-types` returns anything other than 'application/x-apple-diskimage' then we should force `getArchiveMimeType` to return 'application/octet-stream'. (@jviotti does every mime-type returned by `getArchiveMimeType` need a corresponding handler in `lib/image-stream/handlers.js`?)

Alternatively, if all `mime-types` is doing is checking the file-extension, maybe it'd be easier to just check the file extension for `.dmg` ourselves, and not add the extra `mime-types` dependency?